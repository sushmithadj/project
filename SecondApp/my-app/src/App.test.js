import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import UserLogin from './Quiz-App/components/UserLogin'
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'


cleanup()
test('should rednder the login',()=>{
    render(<UserLogin></UserLogin>);
    const value = screen.getByText("start");
    expect(value).toBeInTheDocument()
    

    
 
})

describe("",()=>{
    test("test on rendering  input HomePage by sending dummyprop ", () => {
        render(
         
            <UserLogin
              userName={"sushma"}
              selectedLang={"english"}
            ></UserLogin>
        
        );
    
        expect(screen.getByDisplayValue("sushma")).toBeInTheDocument();
        expect(screen.getByDisplayValue("english")).toBeInTheDocument();

        const tree = renderer
      .create(
        
          <UserLogin
           userName={"varun"}
           selectedLang={"english"}
          ></UserLogin>
        
      )
      .toJSON();
    console.log("::", tree);
    expect(tree).toMatchSnapshot();
})
});


