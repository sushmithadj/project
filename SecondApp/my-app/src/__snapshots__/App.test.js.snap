// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test on rendering  input HomePage by sending dummyprop  1`] = `
<div
  className="wrapper"
>
  <div
    className="form"
  >
    <input
      autoComplete="off"
      name="name"
      onChange={[Function]}
      type="text"
      value="varun"
    />
    <select
      id="category"
      name="category"
      onChange={[Function]}
    >
      <option
        value="male"
      >
        male
      </option>
      <option
        value="female"
      >
        female
      </option>
      <option
        value="others"
      >
        others
      </option>
    </select>
    <select
      id="languages"
      name="languages"
      onChange={[Function]}
      value="english"
    >
      <option
        value="english"
      >
        english
      </option>
      <option
        value="kannada"
      >
        kannada
      </option>
      <option
        value="hindi"
      >
        hindi
      </option>
    </select>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      start
    </button>
  </div>
</div>
`;
